### PR TITLE
Documentation for HttpData

### PR DESCRIPTION
**Pull request by dinojohn**
Documentation for HttpData

### CODE REVIEW

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 1]
```diff
@@ -1 +1,59 @@
-# Work in progress
\ No newline at end of file
+# HttpData
+`HttpData` is a domain to model the content that is to be written in HttpChannel. `HttpRequest` and `HttpResponse` store content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response for ZIO HTTP.
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]:
```suggestion
`HttpData` is a domain to model the content that is to be written on the Netty's HttpChannel.
```

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 2]
```diff
@@ -1 +1,59 @@
...
+```
+## Client-side usage of `HttpData`
+On the client-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Request`. You may replace it with another `HttpData`.
+
+To attach data to a request you can use the `Response` constructors.
+```scala
+  val req: Request = Request(data = HttpData.fromString("Some String"))
+```
+## Creating an HttpData
+`HttpData` can be created using a lot of constructors.
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]: This line can be removed.

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 3]
```diff
@@ -1 +1,59 @@
-# Work in progress
\ No newline at end of file
+# HttpData
+`HttpData` is a domain to model content for `Requests` and `Response`. ZIO HTTP uses Netty at it's core and Netty handles content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response.
+## Server-side usage of `HttpData`
+On the server-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Response`. You may replace it with another `HttpData`.
+
+To attach data to a response you can use the `Response` constructors.
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]:
```suggestion
To add data to a response you can use the `Response` constructors that takes `HttpData.Empty` as the default value
```

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 4]
```diff
@@ -1 +1,59 @@
...
+```scala
+  val res: Response = Response( data = HttpData.fromString("Some String"))
+```
+## Client-side usage of `HttpData`
+On the client-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Request`. You may replace it with another `HttpData`.
+
+To attach data to a request you can use the `Response` constructors.
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]: Same as above.

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 5]
```diff
@@ -1 +1,59 @@
...
+```scala
+  val req: Request = Request(data = HttpData.fromString("Some String"))
+```
+## Creating an HttpData
+`HttpData` can be created using a lot of constructors.
+### Creating an empty HttpData
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]: We can remove this para

**Review by ShrutiVerma97 — CHANGES_REQUESTED** [review]

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 6]
```diff
@@ -1 +1,53 @@
...
+```scala
+  val streamHttpData: HttpData = HttpData.fromStream(ZStream("a", "b", "c"), CharsetUtil.UTF_8)
+```
+### Creating an HttpData from a `File`
+To create an `HttpData` that encodes a File you can use `HttpData.fromFile`.
+```scala
+  val fileHttpData: HttpData = HttpData.fromFile(new io.File(getClass.getResource("/fileName.txt").getPath))
+```
+## Converting `HttpData` to `ByteBuf`
+To convert an `HttpData` to `ButrBuf`  you can call the `toButeBuf` method on it, which returns a `Task[ByteBuf]`.
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]:
```suggestion
To convert an `HttpData` to `ByteBuf`  you can call the `toButeBuf` method on it, which returns a `Task[ByteBuf]`.
```

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 7]
```diff
@@ -1 +1,53 @@
...
+`HttpData` is a domain to model content for `Requests` and `Response`. ZIO HTTP uses Netty at it's core and Netty handles content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response.
+## Server-side usage of `HttpData`
+On the server-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Response`. You may replace it with another `HttpData`.
+
+To attach data to a response you can use the `Response` constructors.
+```scala
+  val res: Response = Response( data = HttpData.fromString("Some String"))
+```
+## Client-side usage of `HttpData`
+On the client-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Request`. You may replace it with another `HttpData`.
```
Comment (ShrutiVerma97) [inline docs/website/docs/v1.x/dsl/http-data/index.md]: we can mention, for testing purposes we can use.

File: docs/website/docs/v1.x/dsl/http-data/index.md [hunk 8]
```diff
@@ -1 +1,57 @@
-# Work in progress
\ No newline at end of file
+# HttpData
+`HttpData` is a domain to model content for `Request`, `Response` and `ClientRequest`. ZIO HTTP uses Netty at it's core and Netty handles content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response.
+## Server-side usage of `HttpData`
+On the server-side, `ZIO-HTTP` models content in `Request` and `Response` as `HttpData` with `HttpData.Empty` as the default value.
+To attach content while creating a `Response` you can use the `Response` constructor.
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/http-data/index.md]:
```suggestion
To add content while creating a `Response` you can use the `Response` constructor.
```

**Review by amitksingh1490 — APPROVED** [review]

**Review by ShrutiVerma97 — APPROVED** [review]

**Pull request merged**